public/app/features/dashboard-scene/pages/DashboardScenePage.tsx

```diff
@@ -78,7 +78,7 @@ export function DashboardScenePage({ match, route, queryParams, history }: Props
   // Do not render anything when transitioning from one dashboard to another
   if (
     match.params.type !== 'snapshot' &&
-    dashboard.state.uid &&
+    match.params.uid &&
     dashboard.state.uid !== match.params.uid &&
     route.routeName !== DashboardRoutes.Home
   ) {
```

Comment:
this condition is designed to make sure we do not render a dashboard that is not valid for the current URL. 

but when going from Home route (what empty dashboard.state.uid) to a normal route this condition does not catch that scenario as dashboard.state.uid is empty and we only compare ` dashboard.state.uid !== match.params.uid ` if dashboard.state.uid has a value, so if we instead check if ` match.params.uid` has a value and then compare 
` dashboard.state.uid !== match.params.uid `  we should cover this scenario as well